io/io/src/RRawFileUnix.cxx

```diff
@@ -115,8 +115,21 @@ void ROOT::Internal::RRawFileUnix::ReadVImpl(RIOVec *ioVec, unsigned int nReq)
 {
 #ifdef R__HAS_URING
    if (RIoUring::IsAvailable()) {
-      // todo(max) actually use the ring
-      RRawFile::ReadVImpl(ioVec, nReq);
+      RIoUring ring(128);
```

Comment:
Let's make the `RIoUring` class responsible for choosing a ring size.

Reply:
This is done in 72e3936. The ring default is 1024 and will step down by powers of 2 if memory allocation fails. I left the constructor that takes a queue depth, in case users would like to specify that themselves.